chain/network/build.rs

```diff
@@ -0,0 +1,3 @@
+fn main() -> std::io::Result<()> {
+    prost_build::compile_protos(&["src/network_protocol/network.proto"], &["src/"])
```

Comment:
https://github.com/tokio-rs/prost/issues/182 is a nasty failure mode I had encountered in the past with `prost` but it seems that they no longer execute arbitrary code downloaded off internet at build time by default. I'd still petition for vendoring. `build.rs` is a terrible horrible thing.

Reply:
FYI protoc-bin-vendored doesn't make much sense, because it is dynamically linked. We would have to create our own package with statically linked protoc for that.

Reply:
Err, I meant checking in the generated sources, my bad.

Reply:
To unblock this, let's maybe leave this thing as is? "Do what the docs suggest, even if you think you know better" is a strong heuristic. We can revisit this later. 

If we do revisit:

- I'd be against adding binaries: I'd rather have `protoc` in `PATH` than in the repo :-)
- I do like checking in generated sources, provided that there's a test verifying the generated sources are fresh. But that's a contrarian opinion which is a bit exhaustive to argue :-)

Reply:
aaand, it turns out that the thing is worse than I expected.
prost-build (the package needed for code generation) actually checks for the presence of protoc in the system during build of the package itself and panics if it doesn't find it. I had to make the dependency on protoc optional and use conditional compilation to make it work :/